context/app/static/js/components/home/HuBMAPDatasetsChart/HuBMAPDatasetsChart.tsx

```diff
@@ -0,0 +1,202 @@
+import React, { ComponentProps } from 'react';
+import Paper from '@mui/material/Paper';
+
+import ChartWrapper from 'js/shared-styles/charts/ChartWrapper';
+import { useSelectedDropdownIndex } from 'js/shared-styles/dropdowns/DropdownListbox';
+import HorizontalStackedBarChart from 'js/shared-styles/charts/HorizontalStackedBarChart';
+import { useChartPalette } from 'js/shared-styles/charts/HorizontalStackedBarChart/hooks';
+import { ChartArea } from 'js/shared-styles/charts/HorizontalStackedBarChart/style';
+import { useBandScale, useLinearScale, useOrdinalScale } from 'js/shared-styles/charts/hooks';
+import Skeleton from '@mui/material/Skeleton';
+import { TooltipData } from 'js/shared-styles/charts/types';
+import Typography from '@mui/material/Typography';
+import { AnyD3Scale } from '@visx/scale';
+import { getSearchURL } from 'js/components/organ/utils';
+import HuBMAPDatasetsChartDropdown from '../HuBMAPDatasetsChartDropdown';
+import {
+  assayTypeQuery,
+  donorSexQuery,
+  analyteClassQuery,
+  processingStatusQuery,
+  AssaysQueryKey,
+  DonorSexQueryKey,
+  AnalyteClassQueryKey,
+  ProcessingStatusQueryKey,
+  donorRaceQuery,
+  DonorRaceQueryKey,
+} from './queries';
+import {
+  AggregatedDatum,
+  getKeysFromAggregatedData,
+  useAggregatedChartData,
+  useDatasetTypeMap,
+  useOrganOrder,
+  useSearchDataRange,
+} from './hooks';
+
+const margin = { top: 40, right: 50, bottom: 64, left: 100 };
+
+const getOrgan = (d: AggregatedDatum) => d.organ as string;
+
+function HuBMAPDatasetsChartTooltip({ tooltipData }: { tooltipData: TooltipData<AggregatedDatum> }) {
+  if (!tooltipData.bar || !tooltipData.key) return null;
+  return (
+    <>
+      <Typography variant="subtitle2" color="secondary">
+        {tooltipData.bar.data.organ}
+      </Typography>
+      <Typography>{tooltipData.key}</Typography>
+      <Typography variant="h3" component="p" color="textPrimary">
+        {tooltipData.bar.data[tooltipData.key]}
+      </Typography>
+    </>
+  );
+}
+
+interface ColorOption {
+  key: string;
+  data: AggregatedDatum[];
+  dropdownLabel: string;
+  keys: string[];
+  getBarHref?: ComponentProps<typeof HorizontalStackedBarChart<AggregatedDatum, AnyD3Scale, AnyD3Scale>>['getBarHref'];
+}
+
+function HuBMAPDatasetsChart() {
+  const colors = useChartPalette();
+  const [selectedColorDataIndex, setSelectedColorDataIndex] = useSelectedDropdownIndex(0);
+
+  const { organOrder } = useOrganOrder();
+
+  const assayBuckets = useAggregatedChartData<AssaysQueryKey>(assayTypeQuery);
+  const donorSexBuckets = useAggregatedChartData<DonorSexQueryKey>(donorSexQuery);
+  const donorRaceBuckets = useAggregatedChartData<DonorRaceQueryKey>(donorRaceQuery);
+  const analyteClassBuckets = useAggregatedChartData<AnalyteClassQueryKey>(analyteClassQuery);
+  const processingStatusBuckets = useAggregatedChartData<ProcessingStatusQueryKey>(processingStatusQuery);
```

Comment:
Are we fetching the buckets for each view upfront?

Reply:
I settled on this approach so that all the different label keys can be provided to the chart wrapper. This allows us to prevent the legend from resizing when changing between bucket sets.